zeeguu_api/api/user_languages.py

```diff
@@ -0,0 +1,137 @@
+import flask
+import zeeguu
+from flask import request
+from zeeguu.model.language import Language
+from zeeguu.model.user_language import UserLanguage
+
+
+from .utils.route_wrappers import cross_domain, with_session
+from .utils.json_result import json_result
+from . import api
+
+session = zeeguu.db.session
+
+USER_LANGUAGES = "user_languages"
+ADD_USER_LANGUAGE = "user_languages/add"
+DELETE_USER_LANGUAGE = "user_languages/delete"
+INTERESTING_LANGUAGES = "user_languages/interesting"
+READING_LANGUAGES = "user_languages/reading"
+
+
+# ---------------------------------------------------------------------------
+@api.route(f"/{ADD_USER_LANGUAGE}", methods=("POST",))
+# ---------------------------------------------------------------------------
+@cross_domain
+@with_session
+def add_user_language():
+    """
+    This endpoint is for adding a user language.
+    It tries to find the user_language, and otherwise create it.
+    After that set it to reading_news so it appears in the reader.
+
+    :return: "OK" in case of success
+    """
+    language_id = int(request.form.get('language_id', ''))
+
+    language_object = Language.find_by_id(language_id)
+    user_language = UserLanguage.find_or_create(session, flask.g.user, language_object)
+    user_language.set_reading_news()
+    session.add(user_language)
+    session.commit()
+
+    return "OK"
+
+
+# ---------------------------------------------------------------------------
+@api.route(f"/{DELETE_USER_LANGUAGE}/<language_id>", methods=("GET",))
+# ---------------------------------------------------------------------------
+@cross_domain
+@with_session
+def delete_user_language(language_id):
+    """
+    A user can delete a language with a given ID.
+    At the moment this endpoint is used for deleting it from the reader,
```

Comment:
good idea.
but then do a delete language :)

the functionality here will be subsumed by the `modify_user_language`